crates/fuel-core/src/executor.rs

```diff
@@ -157,7 +184,21 @@ where
         &self,
         block: ExecutionBlock,
     ) -> ExecutorResult<ExecutionResult> {
-        let (result, db_transaction) = self.execute_without_commit(block)?.into();
+        let component = match block {
+            ExecutionBlock::DryRun(block) => ExecutionTypes::DryRun(Components {
+                header_to_produce: block.header,
+                transactions_source: OnceTransactionsSource::new(block.transactions),
+                gas_limit: u64::MAX,
+            }),
+            ExecutionBlock::Production(block) => ExecutionTypes::Production(Components {
+                header_to_produce: block.header,
+                transactions_source: OnceTransactionsSource::new(block.transactions),
+                gas_limit: u64::MAX,
+            }),
+            ExecutionBlock::Validation(block) => ExecutionTypes::Validation(block),
+        };
+
```

Comment:
nit: maybe this From<ExecutionBlock> could be extracted

Reply:
It is only used once here for unit tests, and it is better to do that explicitly because it does not apply to other places=)